src/everwillow/statelib/state.py

```
@@ -1,929 +1,491 @@
-    return key_children, metadata
+    Raises:
+        KeyError: If ``updates`` includes a key that is not present in ``state``.
 
+    Examples:
+        >>> base = State.from_pytree({"a": 1, "b": 2})
+        >>> update(base, {("b",): 99}).to_dict()
+        {('a',): 1, ('b',): 99}
+    """
+    if not isinstance(state, State):
+        msg = "Can only update State types"  # type: ignore[unreachable]
+        raise ValueError(msg)
```

Ruff (0.14.2)

Prefer `TypeError` exception for invalid type

(TRY004)